src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -184,51 +237,49 @@ private static void ValidateFrameworksOptionsExistInAssetsFile(LockFile assetsFi
         }
 
         /// <summary>
-        /// Validates and returns the assets file for the given project.
+        /// Returns the path to an assets file for the given project.
         /// </summary>
         /// <param name="project">Evaluated MSBuild project</param>
         /// <param name="logger">Logger for the 'why' command</param>
         /// <returns>Assets file for the given project. Returns null if there was any issue finding or parsing the assets file.</returns>
-        private static LockFile? GetProjectAssetsFile(Project project, ILoggerWithColor logger)
+        private static LockFile? GetProjectAssetsFile(string assetsFilePath, string? projectPath, ILoggerWithColor logger)
         {
-            if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
-            {
-                logger.LogError(
-                    string.Format(
-                        CultureInfo.CurrentCulture,
-                        Strings.Error_NotPRProject,
-                        project.FullPath));
-
-                return null;
-            }
-
-            string assetsPath = project.GetPropertyValue(ProjectAssetsFile);
-
-            if (!File.Exists(assetsPath))
+            if (!File.Exists(assetsFilePath))
             {
                 logger.LogError(
                     string.Format(
                         CultureInfo.CurrentCulture,
                         Strings.Error_AssetsFileNotFound,
-                        project.FullPath));
+                        assetsFilePath));
```

Comment:
If we are passing assersFilePath as an argument to the error instead of project.FullPath, we should update the `Error_AssersFileNotFound` to follow suit as well.

Reply:
Good find!